edgelet/iotedged/src/lib.rs

```diff
@@ -392,14 +402,21 @@ where
                 match manual.authentication_method() {
                     ManualAuthMethod::DeviceConnectionString(cs) => {
                         info!("Starting provisioning edge device via manual mode using a device connection string...");
+                        let (key, device_id, hub) = cs
+                            .parse_device_connection_string()
+                            .context(ErrorKind::Initialize(InitializeErrorReason::LoadSettings))?;
+                        let manual = ManualProvisioning::new(key, device_id, hub);
+
                         let (key_store, provisioning_result, root_key) =
-                            manual_provision_connection_string(&cs, &mut tokio_runtime)?;
+                            manual_provision_connection_string(manual.clone(), &mut tokio_runtime)?;
```

Comment:
I thought we'd discussed that you could change `Provision::provision` to take `&self` so that you didn't have to `clone()` the `Provision` impls. Did that not work out?

Reply:
Had forgotten about it, changed it now.